packages/core/data-transfer/src/strapi/remote/controllers/push.ts

```diff
@@ -91,29 +95,33 @@ const createPushController = (options: ILocalStrapiDestinationProviderOptions):
       },
 
       async assets(payload) {
+        // TODO: close the stream upong receiving an 'end' event instead
         if (payload === null) {
           streams.assets?.end();
           return;
         }
 
-        const { step, assetID } = payload;
+        const { action, assetID } = payload;
 
         if (!streams.assets) {
           streams.assets = await provider.getAssetsStream();
         }
 
-        if (step === 'start') {
+        if (action === 'start') {
           assets[assetID] = { ...payload.data, stream: new PassThrough() };
           writeAsync(streams.assets, assets[assetID]);
         }
 
-        if (step === 'stream') {
-          const chunk = Buffer.from(payload.data.chunk.data);
+        if (action === 'stream') {
+          // The buffer has gone through JSON operations and is now of shape { type: "Buffer"; data: UInt8Array }
+          // We need to transform it back into a Buffer instance
+          const rawBuffer = payload.data as unknown as { type: 'Buffer'; data: Uint8Array };
```

Comment:
Just curious, what's the risk of something going wrong here / how would an error be handled?

Reply:
Something would go wrong if the received payload.data isn't a stringified then parsed Buffer instance.
So it's unlikely as long as we're the ones sending the packets. However, it's not impossible at all. In case of an error, It should propagate the error to the WS callback and send an error message to the client.
I didn't touched the data validation for now because I believe it should be a separate topic (validation every message through yup, joi, or smth like that.